arc.go

```diff
@@ -78,28 +95,32 @@ func (fsm *ArcFSM) Insert(ctx context.Context, dbc *sql.DB, st Status, inserter
 		}
 	}
 	if !found {
-		return 0, errors.Wrap(ErrInvalidStateTransition, "invalid insert status and inserter", j.KV("status", st.ShiftStatus()))
+		return 0, nil, errors.Wrap(ErrInvalidStateTransition, "invalid insert status and inserter", j.KV("status", st.ShiftStatus()))
 	}
 
+	return insertTx(ctx, tx, st, inserter, fsm.events, reflex.EventType(st), fsm.options)
+}
+
+func (fsm *ArcFSM) Update(ctx context.Context, dbc *sql.DB, from, to Status, updater Updater) error {
 	tx, err := dbc.Begin()
 	if err != nil {
-		return 0, err
+		return err
 	}
 	defer tx.Rollback()
 
-	id, notify, err := insertTx(ctx, tx, st, inserter, fsm.events, reflex.EventType(st), fsm.options)
+	notify, err := fsm.UpdateTx(ctx, tx, from, to, updater)
 	if err != nil {
-		return 0, err
+		return err
 	}
 	defer notify()
```

Comment:
Should we not notify only if the tx.Commit() succeeds?

Reply:
Yes, now done, and the same for ARC in shift.go.